addons/html_editor/static/tests/toolbar.test.js

```diff
@@ -18,3 +33,29 @@ test("toolbar works: can format bold", async () => {
     await contains(".btn[name='bold']").click();
     expect(getContent(el)).toBe("<p><strong>[test]</strong></p>");
 });
+
+test("toolbar buttons react to selection change", async () => {
+    const { el } = await setupEditor("<p>test some text</p>");
+
+    // set selection to open toolbar
+    setRange(el, "<p>[test] some text</p>");
+    await waitFor(".o-we-toolbar");
+
+    // check that bold button is not active
+    expect(".btn[name='bold']").not.toHaveClass("active");
+
+    // click on toggle bold
+    await contains(".btn[name='bold']").click();
+    expect(getContent(el)).toBe("<p><strong>[test]</strong> some text</p>");
+    expect(".btn[name='bold']").toHaveClass("active");
+
+    // set selection where text is not bold
+    setRange(el, "<p><strong>test</strong> some [text]</p>");
+    await waitFor(".btn[name='bold']:not(.active)");
```

Comment:
i think i would just do a await nextTick()

Reply:
@ged-odoo 
One tick is not enough... (because of Owl's async rendering?)
A longer timeout like `await new Promise(resolve => setTimeout(resolve, 50));` does work...